src/transaction_processor.py

```diff
@@ -134,9 +132,8 @@ def process_single_transaction(
             # transaction_tokens = self.blockchain_data.get_transaction_tokens(tx_hash)
             # store transaction tokens
             transaction_tokens = []
-            for token_address, imbalance in token_imbalances.items():
-                if imbalance != 0:
-                    transaction_tokens.append((tx_hash, token_address))
+            for token_address in token_imbalances.keys():
```

Comment:
I would reorganize code here.

Instead of fetching `token_imbalances` in the beginning, and then having this loop here, both should be moved into a `self.imbalances.get_transaction_tokens` function. That function would then be called here. The function `self.blockchain_data.get_transaction_tokens` should then be deleted.

Reply:
As it is still not clear where and how the raw imbalances will be used, i decided to keep it separate and not part of a get_transaction_tokens function.

Rearranged a few things and modified the test as well (not sure if you like how it looks like), in order to check the current version of the code.